docs/TESTS.md

```diff
@@ -1,28 +1,112 @@
 
+## Running Tests
+
+###### If the exercise's folder looks like this...
+
+```
+./exercise-name_test.hs
+```
+
+Execute the following command to run the tests:
+
 ```bash
-$ runhaskell -Wall bob_test.hs
+stack runghc exercise-name_test.hs
+```
+
+###### Otherwise, it will be like this...
+
+```
+./stack.yaml
+./package.yaml
+./src/ModuleName.hs
+./test/Test.hs
 ```
 
-## Making Your First Haskell Module
+In which case you should type:
+
+```bash
+stack test
+```
+
+## Solving the exercise
+
+A solution to an exercise is a Haskell module. The module's name is defined
+in an *import statement* in the test suite, usually at the beginning:
+
+```haskell
+import Control.Monad (unless)
+import System.Exit   (exitFailure)
+
+import Test.HUnit
+
+import ModuleName (someFunc)
 
-To create a module that can be loaded with `import Bob (responseFor)`, put this code in `Bob.hs`:
+main :: IO ()
+main ...
+```
+
+In this example, the test file imports, in line 6, a module named
+`ModuleName`, that exports a function named `someFunc`. This means you have
+to create a file named `ModuleName.hs` that would be something like this:
 
 ```haskell
-module Bob (responseFor) where
+module ModuleName (someFunc) where
 
-responseFor :: String -> String
-responseFor = undefined
+someFunc :: ...
+someFunc ...
 ```
 
-## (Optional) HLint
+If you are lucky, you'll a find a file with this name already in place - in
```

Comment:
This may imply that "if I retry to fetch the exercise again, maybe I will get lucky this time", so maybe just "If the exercise provides it" or something